src/api/layer.js

```diff
@@ -157,16 +148,11 @@ export default class Layer {
      */
     setStyle(style) {
         this._checkStyle(style);
-        return this._styleChanged(style).then(r => {
-            if (this._style) {
-                this._style.onChange(null);
-            }
-            this._style = style;
-            this._style.onChange(() => {
-                this._styleChanged(style);
-            });
-            return r;
-        });
+        this._style.onChange(null);
```

Comment:
What if there is no style?

Reply:
It fails but it's impossible since the constructor forces you to set a source and a style

Reply:
But the constructor set it asynchronously, right?